### ISSUE TITLE
VS 2017 15.4 Crashes while editing .cs file

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#22886]
Apologies for the cross-post, but I am unable to tie this to the existing problem: https://developercommunity.visualstudio.com/content/problem/133454/vs-2017-154-crashes-while-editing-cs-file.html and I am unable to upload the crash dump file to a comment on the original post as requested by Jinu Joseph [MSFT].

I have installed version 15.4.1 as requested and still see the issue reported earlier.

Some more background. On smaller solutions (ones with a handful of projects) I am unable to replicate the issue.

The solution where I continue to experience this pain has ~ 20 projects in it.

The issue happens like clockwork if I declare a method parameter based on a custom enum or type in our project. 

VS will allow me to specify the type of the parameter, but as soon as I press the space key and begin to type a name for the new method parameter VS crashes. 

Is there a way to turn off the feature that attempts to suggest names of methods in Visual Studio. This might be anectdotal evidence, but I didn't have this issue before that feature dropped (which maybe was in 15.3?)

Attached is the dmp file from the latest crash.​

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/137578/vs-2017-154-crashes-while-editing-cs-file-1.html
VSTS ticketId: 514286_
_These are the original issue comments:_

Jinu Joseph [MSFT] on ‎10‎/‎26‎/‎2017, 06:11 PM (14 hours ago): <p>We appreciate you taking the time to report this problem. We are currently prioritizing problems that are impacting a broad set of our customers, so we may not be able to investigate this one immediately. We know this problem is important to you, so we will continue to monitor it.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by sharwell** [comment]
The stack trace from the heap dump indicates a problem in the error handling for naming recommendations + naming styles:

```
Exception type:   System.InvalidOperationException
Message:          Sequence contains no elements
InnerException:   <none>
StackTrace (generated):
    SP       IP       Function
    3976E5D8 2AC36E55 System_Core_ni!System.Linq.Enumerable.Single[[Microsoft.CodeAnalysis.NamingStyles.NamingStyle, Microsoft.CodeAnalysis.Workspaces]](System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.NamingStyles.NamingStyle>)+0x185
    3976E64C 57844C86 System_Collections_Immutable_ni!System.Linq.ImmutableArrayExtensions.Single[[Microsoft.CodeAnalysis.NamingStyles.NamingStyle, Microsoft.CodeAnalysis.Workspaces]](System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.NamingStyles.NamingStyle>, System.Func`2<Microsoft.CodeAnalysis.NamingStyles.NamingStyle,Boolean>)+0x12e
    3976E6E4 57832B98 Microsoft_CodeAnalysis_Workspaces_ni!Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingStylePreferences.GetNamingStyle(System.Guid)+0x54
    3976E710 578A36E3 Microsoft_CodeAnalysis_Workspaces_ni!Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SerializableNamingRule.GetRule(Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingStylePreferences)+0x63
    3976E7A4 578A3791 Microsoft_CodeAnalysis_Workspaces_ni!Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingStylePreferences.<CreateRules>b__14_0(Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SerializableNamingRule)+0x11
    3976E7AC 2AC36C0A System_Core_ni!System.Linq.Enumerable+WhereSelectArrayIterator`2[[System.__Canon, mscorlib],[Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule, Microsoft.CodeAnalysis.Workspaces]].MoveNext()+0x4a
    3976E7BC 2AC36AFC System_Core_ni!System.Linq.Buffer`1[[Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule, Microsoft.CodeAnalysis.Workspaces]]..ctor(System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule>)+0x124
    3976E81C 2AC3697E System_Core_ni!System.Linq.Enumerable.ToArray[[Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule, Microsoft.CodeAnalysis.Workspaces]](System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule>)+0x2e
    3976E83C 57844AD1 System_Collections_Immutable_ni!System.Collections.Immutable.ImmutableArray.CreateRange[[Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule, Microsoft.CodeAnalysis.Workspaces]](System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule>)+0xa9
    3976E850 57844A25 System_Collections_Immutable_ni!System.Collections.Immutable.ImmutableArray.ToImmutableArray[[Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule, Microsoft.CodeAnalysis.Workspaces]](System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingRule>)+0x41
    3976E858 578200C2 Microsoft_CodeAnalysis_Workspaces_ni!Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.NamingStylePreferences.CreateRules()+0x5e
    3976E870 3E86A108 Microsoft_CodeAnalysis_CSharp_Features_ni!Microsoft.CodeAnalysis.CSharp.Completion.Providers.DeclarationNameCompletionProvider+<GetRecommendedNamesAsync>d__13.MoveNext()+0x180
    3976EA0C 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976EA1C 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976EA28 676FFB39 mscorlib_ni!System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[[System.__Canon, mscorlib]].GetResult()+0x1d
    3976EA30 29A63DCA Microsoft_CodeAnalysis_CSharp_Features_ni!Microsoft.CodeAnalysis.CSharp.Completion.Providers.DeclarationNameCompletionProvider+<ProvideCompletionsAsync>d__9.MoveNext()+0x44a
    3976EBC0 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976EBD0 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976EBDC 57009041 Microsoft_CodeAnalysis_Features_ni!Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders+<GetContextAsync>d__31.MoveNext()+0x131
    3976ED44 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976ED54 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976ED60 57008DED Microsoft_CodeAnalysis_Features_ni!Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders+<ComputeNonEmptyCompletionContextsAsync>d__24.MoveNext()+0x16d
    3976EEA4 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976EEB4 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976EEC0 570086BC Microsoft_CodeAnalysis_Features_ni!Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders+<GetCompletionsAsync>d__22.MoveNext()+0x3fc
    3976F058 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976F068 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976F074 56B204A2 Microsoft_CodeAnalysis_EditorFeatures_ni!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Controller+Session+ModelComputer+<GetCompletionListAsync>d__14.MoveNext()+0xb2
    3976F15C 676D4B68 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x64
    3976F16C 676D0817 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x3b
    3976F178 56B1FF7E Microsoft_CodeAnalysis_EditorFeatures_ni!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Controller+Session+ModelComputer+<DoInBackgroundAsync>d__13.MoveNext()+0x25e
```

**Comment by tlentine** [comment]
Is there a suggested workaround to this issue? I hit this multiple times each day as it stands now.

**Comment by sharwell** [comment]
@tlentine Can you post a copy of your .editorconfig here? It sounds like it may contain one or more invalid settings for naming conventions.

**Comment by tlentine** [comment]
Sure!

[editorconfig.zip](https://github.com/dotnet/roslyn/files/1435547/editorconfig.zip)

**Comment by sharwell** [comment]
Content of the file attached to the previous comment:

<details><summary>.editorconfig</summary>

```
# EditorConfig is awesome:http://EditorConfig.org

# top-most EditorConfig file
root = true

# Don't use tabs for indentation.
[*]
indent_style = space
# (Please don't specify an indent_size here; that has too many unintended consequences.)

# Code files
[*.{cs,csx,vb,vbx}]
indent_size = 4

# Xml project files
[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,projitems,shproj}]
indent_size = 2

# Xml config files
[*.{props,targets,ruleset,config,nuspec,resx,vsixmanifest,vsct}]
indent_size = 2

# JSON files
[*.json]
indent_size = 2

# Dotnet code style settings:
[*.{cs,vb}]
# Sort using and Import directives with System.* appearing first
dotnet_sort_system_directives_first = true
# Avoid "this." and "Me." if not necessary
dotnet_style_qualification_for_field = false:suggestion
dotnet_style_qualification_for_property = false:suggestion
dotnet_style_qualification_for_method = false:suggestion
dotnet_style_qualification_for_event = false:suggestion

# Use language keywords instead of framework type names for type references
dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
dotnet_style_predefined_type_for_member_access = true:suggestion

# Suggest more modern language features when available
dotnet_style_object_initializer = true:suggestion
dotnet_style_collection_initializer = true:suggestion
dotnet_style_coalesce_expression = true:suggestion
dotnet_style_null_propagation = true:suggestion
dotnet_style_explicit_tuple_names = true:suggestion

# name all constant fields using PascalCase
dotnet_naming_rule.constant_fields_should_be_pascal_case.severity = suggestion
dotnet_naming_rule.constant_fields_should_be_pascal_case.symbols  = constant_fields
dotnet_naming_rule.constant_fields_should_be_pascal_case.style    = pascal_case_style

dotnet_naming_symbols.constant_fields.applicable_kinds   = field
dotnet_naming_symbols.constant_fields.required_modifiers = const

dotnet_naming_rule.methods_and_properties_must_be_pascal_case.severity = warning
dotnet_naming_rule.methods_and_properties_must_be_pascal_case.symbols  = method_and_property_symbols
dotnet_naming_rule.methods_and_properties_must_be_pascal_case.style    = pascal_case_style

dotnet_naming_symbols.method_and_property_symbols.applicable_kinds = method, property
dotnet_naming_symbols.method_and_property_symbols.applicable_accessibilities = *

dotnet_naming_style.pascal_case_style.capitalization = pascal_case

# internal and private fields should be _camelCase
dotnet_naming_rule.camel_case_for_private_internal_fields.severity = suggestion
dotnet_naming_rule.camel_case_for_private_internal_fields.symbols  = private_internal_fields
dotnet_naming_rule.camel_case_for_private_internal_fields.style    = camel_case_underscore_style

dotnet_naming_symbols.private_internal_fields.applicable_kinds = field
dotnet_naming_symbols.private_internal_fields.applicable_accessibilities = private, internal

# dotnet_naming_style.camel_case_underscore_style.required_prefix = _
# dotnet_naming_style.camel_case_underscore_style.capitalization = camel_case 

# CSharp code style settings:
[*.cs]
# Prefer "var" everywhere
csharp_style_var_for_built_in_types = true:suggestion
csharp_style_var_when_type_is_apparent = true:suggestion
csharp_style_var_elsewhere = true:suggestion


# Prefer method-like constructs to have a block body
csharp_style_expression_bodied_methods = false:none
csharp_style_expression_bodied_constructors = false:none
csharp_style_expression_bodied_operators = false:none

# Prefer property-like constructs to have an expression-body
csharp_style_expression_bodied_properties = true:none
csharp_style_expression_bodied_indexers = true:none
csharp_style_expression_bodied_accessors = true:none

# Suggest more modern language features when available
csharp_style_pattern_matching_over_is_with_cast_check = true:suggestion
csharp_style_pattern_matching_over_as_with_null_check = true:suggestion
csharp_style_inlined_variable_declaration = true:suggestion
csharp_style_throw_expression = true:suggestion
csharp_style_conditional_delegate_call = true:suggestion

# Newline settings
csharp_new_line_before_open_brace = control_blocks:none

csharp_new_line_before_else = false
csharp_new_line_before_catch = true
csharp_new_line_before_finally = true
csharp_new_line_before_members_in_object_initializers = false
csharp_new_line_before_members_in_anonymous_types = false
```

</details>

**Comment by sharwell** [comment]
@tlentine It looks like you reference a naming style `camel_case_underscore_style` which is not defined (the lines are commented out). If you uncomment the two lines that define it, does it prevent the crashes?

**Comment by tlentine** [comment]
@sharwell  That seems to work! Now to run git blame to find out when & who commented those styles out : )

In all seriousness, thanks for pointing this out - this was driving me crazy.

**Comment by rik-smeets** [comment]
I recently ran into this problem too, so I looked into it and found it has already been resolved with [commit 32b9d5f5](https://github.com/dotnet/roslyn/commit/32b9d5f5), which has been merged to master in #22113. 

It did not release yet, so if someone reading this is still having this issue, check your .editorconfig file for inconsistencies like the one mentioned above. Fixing them will resolve the issue.
